src/inmanta/env.py

```diff
@@ -101,6 +101,36 @@ def _init_env(self) -> bool:
                 return False
             LOGGER.debug("Created a new virtualenv at %s", self.env_path)
 
+        else:
+            # make sure the venv hosts the same python version as the running process
+
+            if sys.platform != "win32":
```

Comment:
I don't think this is correct. We should perform the `python --version` check on any platform that is not Linux and the faster check on Linux systems. This implementation does it the other way around. It performs the fast check on any platform that is not a windows, but we don't know whether the fast check will work on any non-windows system.

I think the same bug exists on master and iso5, so we will have to patch it there as well.

Reply:
Created a PR here: https://github.com/inmanta/inmanta-core/pull/3992 for master and iso5